apps/aecore/src/aec_chain_state.erl

```diff
@@ -758,7 +758,9 @@ remove_locations(Hash, Hash) ->
 remove_locations(StopHash, CurrentHash) ->
     lists:foreach(fun(TxHash) ->
                           aec_db:remove_tx_location(TxHash),
-                          aec_db:add_tx_hash_to_mempool(TxHash)
+                          aec_db:add_tx_hash_to_mempool(TxHash),
+                          Tx = aec_db:get_signed_tx(TxHash),
```

Comment:
The good thing is that this won't regress on sync performance - this only executes on microforks.